sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -1810,3 +1810,83 @@ case class MakeTimestamp(
 
   override def prettyName: String = "make_timestamp"
 }
+
+@ExpressionDescription(
+  usage = "_FUNC_(date) - Returns the millennium of the date/timestamp.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_('2000-01-19');
+       2
+      > SELECT _FUNC_('2019-08-06');
+       3
+  """,
+  since = "3.0.0")
+case class Millennium(child: Expression) extends UnaryExpression with ImplicitCastInputTypes {
```

Comment:
Could you update the title since this PR adds these function beyond `EXTRACT` syntax?